paddle/cinn/backends/compiler.cc

```diff
@@ -29,13 +30,125 @@
 #endif
 
 DECLARE_string(cinn_source_code_save_path);
+DECLARE_string(cinn_dump_group_lowered_func);
+DECLARE_string(cinn_dump_group_source_code);
+DECLARE_string(cinn_dump_group_ptx);
+DECLARE_string(cinn_dump_group_instruction);
 
 namespace cinn {
 namespace backends {
 using ir::Module;
 
 static constexpr int DebugLogMaxLen = 30000;
 
+void DumpCompilationInfo::DumpLoweredFunc() {
+  if (FLAGS_cinn_dump_group_lowered_func.empty()) {
+    return;
+  }
+  for (int idx = 0; idx < info_.lowered_funcs.size(); idx++) {
+    auto dump_path = utils::StringFormat(
+        "%s/fusion_group_%d", FLAGS_cinn_dump_group_lowered_func.c_str(), idx);
+    if (!hlir::framework::MakeDirectory(
+            dump_path, S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH)) {
+      LOG(WARNING) << "Failed to make directory: \"" << dump_path
+                   << "\", the lowered functions for this group will not dump.";
+    } else {
+      auto dump_file = utils::StringFormat(
+          "%s/%s", dump_path.c_str(), "lowered_function.txt");
+      VLOG(7) << "Dump lower functions to: " << dump_file;
+      std::ofstream of(dump_file, std::ios_base::out);
+      if (of.is_open()) {
+        of << info_.lowered_funcs[idx].front();
+      } else {
+        LOG(WARNING) << "Failed to open file: " << dump_file
+                     << ", please check your path.";
+      }
+      of.close();
```

Comment:
Based on the C++ reference, close on an `ofstream` which is not opened may throw [ios_base::failure](https://cplusplus.com/ios_base::failure)

Could you put `of.close()` in `if (of.is_open())` condition branch to make code safer? Only close after you open it. : -)

Reference: https://cplusplus.com/reference/fstream/ofstream/close/